fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt

```diff
@@ -141,38 +141,55 @@ class ProductAnalysisMapperTest {
     }
 
     @Test
-    fun `WHEN grade, rating and highlights are all null and it is initial analysis THEN it is mapped to no analysis present`() {
-        val actual =
-            ProductAnalysisTestData.productAnalysis(
-                grade = null,
-                adjustedRating = null,
-                highlights = null,
-            ).toProductReviewState(isInitialAnalysis = true)
-        val expected = ReviewQualityCheckState.OptedIn.ProductReviewState.NoAnalysisPresent()
+    fun `WHEN there are not enough reviews and no analysis needed THEN not enough reviews card is visible`() {
```

Comment:
Let's also add one more test where `notEnoughReviews` and `needsAnalysis` are both true, that should show the data with needs analysis status.

Reply:
Good idea, added NeedsAnalysis test